content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/execute.rb

```diff
@@ -16,36 +16,47 @@ def main
                 @handle.log("info", "Starting execute")
 
                 begin
-                  service.execute(@handle.root["service_action"])
+                  service.execute(service_action)
                   @handle.root['ae_result'] = 'ok'
                   @handle.log("info", "Ending execute")
                 rescue => err
                   @handle.root['ae_result'] = 'error'
                   @handle.root['ae_reason'] = err.message
-                  task.miq_request.user_message = err.message
-                  @handle.log("info", "Error in execute")
+                  @handle.log("info", "Error in execute: #{err.message}")
+                  update_task(err.message)
```

Comment:
@tinaafitz 
Should update_task take a second parameter to indicate if its info or error?